### PR TITLE
chore: improve SharedDeckDownloadFragment logging

### PR DESCRIPTION
**Pull request by criticalAY**
## Purpose / Description
The log gives only the status code not the reason, I added the reason too in the log

## How Has This Been Tested?

`Status code -> 16, reason 1008`

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/SharedDecksDownloadFragment.kt [hunk 1]
```diff
@@ -244,11 +244,12 @@ class SharedDecksDownloadFragment : Fragment(R.layout.fragment_shared_decks_down
                     }
 
                     val columnIndex: Int = it.getColumnIndex(DownloadManager.COLUMN_STATUS)
+                    val columnIndexReason: Int = it.getColumnIndex(DownloadManager.COLUMN_REASON)
 
                     // Return if download was not successful.
                     if (it.getInt(columnIndex) != DownloadManager.STATUS_SUCCESSFUL) {
                         Timber.i("Download could not be successful, update UI and unregister receiver")
-                        Timber.d("Status code -> ${it.getInt(columnIndex)}")
+                        Timber.d("Status code -> ${it.getInt(columnIndex)}, reason ${it.getInt(columnIndexReason)}")
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/SharedDecksDownloadFragment.kt:252]: (On my phone), is this nullable, or could it potentially throw

Comment (criticalAY) [reply AnkiDroid/src/main/java/com/ichi2/anki/SharedDecksDownloadFragment.kt:252]: Its not nullable, value is garunteed

Comment (david-allison) [reply AnkiDroid/src/main/java/com/ichi2/anki/SharedDecksDownloadFragment.kt:252]:
```java
     * If {@link #COLUMN_STATUS} is neither {@link #STATUS_FAILED} nor {@link #STATUS_PAUSED}, this
     * column's value is undefined.
```

getInt:

```kotlin
Returns the value of the requested column as an int.
The result and whether this method throws an exception when the column value is null, the column type is not an integral type, or the integer value is outside the range [Integer.MIN_VALUE, Integer.MAX_VALUE] is implementation-defined.
```

write a `getIntOrNull` extension method on `Cursor`

Comment (criticalAY) [reply AnkiDroid/src/main/java/com/ichi2/anki/SharedDecksDownloadFragment.kt]: Sorry :)

File: AnkiDroid/src/main/java/com/ichi2/anki/SharedDecksDownloadFragment.kt [hunk 2]
```diff
@@ -289,6 +291,17 @@ class SharedDecksDownloadFragment : Fragment(R.layout.fragment_shared_decks_down
...
 
+    /**
+     * Safely retrieves the integer value from the cursor at the specified column index.
+     * Returns null if the column index is invalid or the column value is null.
+     *
+     * @param columnIndex The index of the column from which to retrieve the integer value.
+     * @return The integer value from the cursor at the specified column index, or null if invalid or undefined.
+     */
+    fun Cursor?.getIntOrNull(columnIndex: Int): Int? {
+        return if (columnIndex != -1) this?.getInt(columnIndex) else null
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/SharedDecksDownloadFragment.kt:302]: The issue is more likely to be with bad data (null/out of range/non integer), also wrap in a try...catch

**Review by david-allison — CHANGES_REQUESTED** [review]
Last one!

File: AnkiDroid/src/main/java/com/ichi2/anki/SharedDecksDownloadFragment.kt [hunk 3]
```diff
@@ -289,6 +291,24 @@ class SharedDecksDownloadFragment : Fragment(R.layout.fragment_shared_decks_down
         }
     }
 
+    /**
+     * Safely retrieves the integer value from the cursor at the specified column index.
+     *
+     * @param columnIndex The index of the column from which to retrieve the integer value.
+     * @return The integer value from the cursor at the specified column index, or null if invalid or undefined.
+     */
+    fun Cursor?.getIntOrNull(columnIndex: Int): Int? {
```
Comment (lukstbit) [inline AnkiDroid/src/main/java/com/ichi2/anki/SharedDecksDownloadFragment.kt:300]: One small nitpick, but make this method private.

**Review by lukstbit — APPROVED** [review]

**Review by david-allison — APPROVED** [review]

**Pull request merged**